DOCS.md

```diff
@@ -69,270 +70,325 @@ In a Sponsored Ad Break flow, once the user hits a mid-roll break with a true[X]
+This is a [terminal event](#terminal-events). This event will fire when the true[X] unit has encountered an unrecoverable error. The app code should handle this the same way as an `onAdCompleted` event -- resume playback and remove all references to the `TruexAdRenderer` instance.
 
 
-#### onNoAdsAvailable
+#### `onNoAdsAvailable`
 
-    -(void) onNoAdsAvailable
+```objective-c
+    - (void)onNoAdsAvailable;
+```
 
-This event will fire when the true[X] unit has determined it has no ads available to show the current user. The app code should handle this the same way as an `AD_COMPLETED` event -- resume playback and remove all references to the `TruexAdRenderer` instance.
+This is a [terminal event](#terminal-events). This event will fire when the true[X] unit has determined it has no ads available to show the current user. The app code should handle this the same way as an `onAdCompleted` event -- resume playback and remove all references to the `TruexAdRenderer` instance.
 
 
-#### onAdFreePod
+#### `onAdFreePod`
 
-    -(void) onAdFreePod
+```objective-c
+    - (void)onAdFreePod;
+```
 
-This event will fire when the all remaining ads in the current ad pod need to be skipped. The app code should notate that this event has fired, but should not take any further action until it receives an `AD_COMPLETED` or `AD_ERROR` event.
+This event will fire when the user has earned a credit with true[X]. The app code should notate that this event has fired, but should not take any further action. Upon receiving a [terminal event](#terminal-events), if `onAdFreePod` was fired, the app should skip all remaining ads in the current slot. If it was not fired, the app should resume playback without skipping any ads, so the user receives a normal video ad payload.
 
-*All following events are used mostly for tracking purposes -- no action is generally required:*
 
+#### `onUserCancelStream`
 
-#### onOptIn *(optional)*
+```objective-c
+    - (void)onUserCancelStream;
+```
 
-    -(void) onOptIn:(NSString *)campaignName adId:(NSInteger)adId
+This is a [terminal event](#terminal-events), and is marked as **`@optional`** in the `TruexAdRendererDelegate` protocol.
 
-This event will fire if the user selects to interact with the true[X] interactive ad.
+Implementing this delegate method will enable the renderer to fire this event when the user intends to exit the stream entirely. The app, at this point, should treat this the same way it would handle any other "exit" action from within the stream -- in most cases this will result in the user being returned to an episode/series detail page.
 
-Note that this event may be fired multiple times if a user opts in to the true[X] interactive ad and subsequently backs out.
+If the app code does not implement this delegate method, then the renderer will emit an [`onAdCompleted`](#onadcompleted) event instead.
 
 
-#### onOptOut *(optional)*
+### `TruexAdRenderer` Output Events -- Informative
```

Comment:
Can this title be "... -- Informative (Optional)"?

Reply:
I actually put an `@optional` method in the "Main Flow" section. So I thought putting "(Optional)" here would add to confusion.

Unless I've made it more confusing by using "Main Flow" and "Informative"...